src/OpenSage.Game/Game.cs

```diff
@@ -86,6 +87,100 @@ public sealed class Game : DisposableBase
         /// </summary>
         public bool IsRunning { get; }
 
+        public void LoadReplayFile(FileSystemEntry replayFileEntry)
+        {
+
+            ReplayFile replayFile = ReplayFile.FromFileSystemEntry(replayFileEntry);
+
+            // TODO: This probably isn't right.
+            var mapFilenameParts = replayFile.Header.Metadata.MapFile.Split('/');
+            var mapFilename = $"Maps\\{mapFilenameParts[1]}\\{mapFilenameParts[1]}.map";
+
+            // TODO: set the correct factions & colors
+            List<PlayerSetting?> pSettings = new List<PlayerSetting?>();
+
+            foreach(var slot in replayFile.Header.Metadata.Slots)
+            {
+                if(slot.SlotType != ReplaySlotType.Empty)
+                {
+                    var owner = PlayerOwner.Player;
+                    if(slot.SlotType == ReplaySlotType.Computer)
+                    {
+                        switch (slot.ComputerDifficulty)
+                        {
+                            case ReplaySlotDifficulty.Easy:
+                                owner = PlayerOwner.EasyAi;
+                                break;
+
+                            case ReplaySlotDifficulty.Medium:
+                                owner = PlayerOwner.MediumAi;
+                                break;
+
+                            case ReplaySlotDifficulty.Hard:
+                                owner = PlayerOwner.HardAi;
+                                break;
+                        }
+                    }
+                    var factionIndex = slot.Faction;
+                    if(factionIndex == -1) // random
+                    {
+                        var maxFactionIndex = ContentManager.IniDataContext.PlayerTemplates.Count();
+                        var minFactionIndex = 2; //0 and 1 are civilian and observer
+                        var random = new Random();
+
+                        var diff = maxFactionIndex - minFactionIndex;
+                        factionIndex = minFactionIndex + (random.Next() % diff);
+                    }
+
+                    var faction = ContentManager.IniDataContext.PlayerTemplates[factionIndex];
+
+                    Console.WriteLine(ContentManager.IniDataContext.PlayerTemplates);
+                    var color = new ColorRgb(0, 0, 0);
+                  
+                    switch (slot.Color)
```

Comment:
I don't think those are hardcoded. Maybe check in MutiplayerColors or some other ini entry

Reply:
colors get loaded from ini files now :)